search/django/weedid/views.py

```diff
@@ -16,3 +30,139 @@ def elasticsearch_query(request):
         headers=request.headers,
     )
     return HttpResponse(elasticsearch_response)
+
+
+@csrf_exempt
```

Comment:
Why is it exempt?

Reply:
This one I'm not too sure, but without the exemption request would be rejected due to csrf violation.

Reply:
I think it shouldn't be exempt. I think you need to get the CSRF token within the react app and then pass it as the `X-CSRFToken` header in every call to this API. See https://docs.djangoproject.com/en/3.1/ref/csrf/, summarised at https://stackoverflow.com/questions/50732815/how-to-use-csrf-token-in-django-restful-api-and-react. I must admit I don't really understand why the cookie automatically being passed to the header is insufficient.

Reply:
I've checked these links when I implemented it at the first place. The one from stackoverflow wasn't working so I ended up with an exemption. I'll see if there's other approach.

Reply:
Resolved